catroid/src/androidTest/java/org/catrobat/catroid/test/utils/CrashReporterTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+
+import org.catrobat.catroid.ui.SettingsActivity;
+import org.catrobat.catroid.utils.CrashReporter;
+import org.junit.Test;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+public class CrashReporterTest {
+
+	@Test
+	public void testCrashlyticsUninitializedOnAnonymousReportDisabled() {
+		Context context = InstrumentationRegistry.getTargetContext();
+		SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
+		editor.putBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, false);
+		editor.clear();
+		editor.commit();
+		CrashReporter crashReporter = new CrashReporter(context);
+		assertFalse(crashReporter.initialize());
+	}
+
+	@Test
+	public void testCrashlyticsInitializedOnAnonymousReportEnabled() {
+		Context context = InstrumentationRegistry.getTargetContext();
+		SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
+		editor.putBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, true);
+		editor.clear();
+		editor.commit();
+		CrashReporter crashReporter = new CrashReporter(context);
+		assertTrue(crashReporter.initialize());
```

Comment:
Structuring test methods in
```
void testMethod() {
  // setup

  // execute

  // assert
}
```
will often improve the readability of tests. Mind the newlines between each block. Rarely no newline between the setup/execute block and assert block improves readability.